### PR TITLE
`Auth` polishing

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

File: client/src/client.rs [hunk 1]
```diff
@@ -163,13 +163,14 @@ impl RawTx for String {
 }
 
 /// The different authentication methods for the client.
-pub enum Auth<'a> {
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Ord, PartialOrd)]
+pub enum Auth {
     None,
     UserPass(String, String),
-    CookieFile(&'a str),
+    CookieFile(String),
```
Comment (sgeisler) [inline client/src/client.rs]: Since we are talking about a file path, wouldn't `std::path::PathBuf` be a more appropriate type?

**Comment by stevenroose** [comment]
I'm good with the derive. I don't really see why it should be owned. The auth should only be passed into the constructor method and is then not reused.

File: client/src/client.rs (hunk 1 above)
Comment (stevenroose) [reply client/src/client.rs]: If we want it to be owned, I think pathbuf makes sense indeed.

**Comment by sgeisler** [comment]
@stevenroose I have no problem with either the owned or the borrowed version, but having this mix of owned and borrowed variants is weird.

**Comment by dpc** [comment]
> I don't really see why it should be owned. 

In my case, I just pass `url` + `Auth` around (already parsed from cmdline/envvars. etc.) and create `Client` from that when appropriate. https://github.com/dpc/rust-bitcoin-indexer/blob/master/src/prelude.rs#L67

File: client/src/client.rs (hunk 1 above)
Comment (dpc) [reply client/src/client.rs]: Done.

**Comment by stevenroose** [comment]
Let's wait with merging breaking things (I marked them v0.6.0) in case there are non-breaking things that can go in first (so we don't have to bother with version branches etc).

I'm doing some work using methods I haven't used before and bumping into some minor things meanwhile.

This one can go in like this once we are going for v0.6.0, IMO.

**Comment by dpc** [comment]
I don't understand @stevenroose . :) Anyway, please hit the "merge pull request" button when you think it's OK.

**Pull request merged**